### DISCUSSION TITLE
Usage "h" function in examples

### DISCUSSION THREAD

**Discussion by naliferov** [preactjs/preact#3123] [Q&A] [answered]
Hi!

What sense of "h" function there?
![](https://image.prntscr.com/image/JseAZ4zmSb_bkOVUJiA19g.png)
Is it useless import when we talking about jsx file?

**Comment by JoviDeCroock** [comment, marked as answer]
It's a transpilation support, basically JSX isn't valid JS so when your transpiler comes into play this component will be transformed to:

```jsx
// From
const HelloWorld = () => <p>Hello world</p>
// TO
const HelloWorld = () => h('p', {}, 'Hello world');
```

So basically so `h` is defined within the scope

**Comment by naliferov** [comment]
Okey, thanks. I think docs lacks of example like this `render( h(Component, props) )` for server side render not in jsx file.

**Reply by JoviDeCroock** [reply]
I mean, you can server-side render with JSX files, that's a language detail and has no particular relation to where it's being used.
